### PR TITLE
Refactor About Referral Section

### PR DESCRIPTION
**Pull request by tavy88**
### Context

Refactor About Referral Section

### Link to Trello card

https://trello.com/c/uaRxOep8

### CODE REVIEW

**Comment by vassyz** [comment]
[AboutThePersonYouAreReferringSection](https://trello.com/c/uaRxOep8/1344-aboutthepersonyouarereferringsection)

File: app/forms/referral_form.rb [hunk 1]
```diff
@@ -49,55 +49,65 @@ def about_you_section
...
+
+    if referral.from_employer?
+      items.append(Referrals::Sections::ReferralContactDetailsSection.new(referral:))
+    end
+
+    items.append(Referrals::Sections::ReferralAboutTheirRoleSection.new(referral:))
+
+    Referrals::SectionGroup.new(slug: "about_the_person_you_are_referring", items:)
+
+    # ReferralSection
```
Comment (paulodeon) [inline app/forms/referral_form.rb:62]: I guess we can remove the commented stuff now

File: app/forms/referrals/sections/referral_contact_details_section.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
+module Referrals
+  module Sections
+    class ReferralContactDetailsSection < Section
+      def items
+        [
+          Referrals::ContactDetails::EmailForm.new(referral:),
+          Referrals::ContactDetails::TelephoneForm.new(referral:),
+          Referrals::ContactDetails::AddressKnownForm.new(referral:),
+          Referrals::ContactDetails::AddressForm.new(referral:),
```
Comment (paulodeon) [inline app/forms/referrals/sections/referral_contact_details_section.rb:9]: should be `referral.address_known? && Referrals::ContactDetails::AddressForm.new(referral:),` with a `.compact_blank`

File: app/forms/referrals/sections/referral_about_their_role_section.rb [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      def items
+        items = [
+          Referrals::TeacherRole::JobTitleForm.new(referral:),
+          Referrals::TeacherRole::DutiesForm.new(referral:)
+        ]
+
+        if referral.from_employer?
+          items.append(Referrals::TeacherRole::SameOrganisationForm.new(referral:))
+
+          if referral.same_organisation?
```
Comment (paulodeon) [inline app/forms/referrals/sections/referral_about_their_role_section.rb:13]: I think this should be `unless referral.same_organisation?`

File: app/forms/referrals/sections/referral_about_their_role_section.rb [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+            items.append(Referrals::TeacherRole::OrganisationAddressKnownForm.new(referral:))
+            if referral.organisation_address_known?
+              items.append(Referrals::TeacherRole::OrganisationAddressForm.new(referral:))
+            end
+          end
+
+          items.append(Referrals::TeacherRole::StartDateForm.new(referral:))
+          items.append(Referrals::TeacherRole::EmploymentStatusForm.new(referral:))
+          items.append(Referrals::TeacherRole::EndDateForm.new(referral:))
```
Comment (paulodeon) [inline app/forms/referrals/sections/referral_about_their_role_section.rb:22]: this should be in the block below

**Pull request merged**